python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,519 @@
+        # schema.
+        afwTable.CoordKey.addErrorFields(self.schema)
+
+        # Standard subtasks to run in sequence.
+        self.subtaskNames = ["detection", "deblend", "measurement"]
+
+    def makeSubtasks(self):
+        """Construct subtasks based on the current configuration."""
+        for name in self.subtaskNames:
+            if not hasattr(self, name):
+                self.makeSubtask(name, schema=self.schema)
+
+    def run(
+        self, exposure: afwImage.Exposure, idGenerator: measBase.IdGenerator = None
+    ) -> afwTable.SourceCatalog:
+        """Run detection, optional deblending, and measurement on a given
+        image.
+
+        Parameters
+        ----------
+        exposure :
+            The exposure on which to detect, deblend and measure sources.
+        idGenerator : optional
+            Object that generates source IDs and provides random seeds.
+
+        Returns
+        -------
+        catalog :
+            The source catalog with all requested measurements.
+        """
+        # Make the `deblend` subtask only if it is enabled.
+        if self.config.deblender is None:
+            self.subtasks.remove("deblend")
+
+        # Validate the configuration.
+        self.config.validate()
+
+        # Ensure this method picks up the current subtask config.
+        self.makeSubtasks()
+        # N.B. subtasks must be created here to handle reconfigurations, such
+        # as retargeting the `deblend` subtask, because the `makeSubtask`
+        # method locks in its config just before creating the subtask. If the
+        # subtask was already made in __init__ using the initial config, it
+        # cannot be retargeted now because retargeting happens at the config
+        # level, not the subtask level.
+
+        if idGenerator is None:
+            idGenerator = measBase.IdGenerator()
+
+        self.exposure = exposure
+
+        # Create an empty source table with the known schema into which
+        # detections will be placed next.
+        self.catalog = afwTable.SourceTable.make(self.schema, idGenerator.make_table_id_factory())
+
+        # Step 1: Detect sources in the image and populate the catalog.
+        self._detectSources()
+
+        # Step 2: If enabled, deblend detected sources and update the catalog.
+        if self.config.deblender:
+            self._deblendSources()
+        else:
+            self.log.info("Deblending is disabled; skipping deblending")
+
+        # Step 3: Measure properties of detected/deblended sources.
+        self._measureSources()
+
+        return self.catalog
+
+    def _detectSources(self):
+        """Run the detection subtask to identify sources in the image."""
+        self.log.info(f"Running detection on a {self.exposure.width}x{self.exposure.height} pixel exposure")
+        self.catalog = self.detection.run(self.catalog, self.exposure).sources
+
+    def _deblendSources(self):
```

Comment:
I would make _deblendSources an abstract method and override it in the single-band and multi-band versions.